coreapi/v24.0/consensus/cometbft/api/api.go

```diff
@@ -0,0 +1,70 @@
+package api
+
+import (
+	"fmt"
+
+	cmtproto "github.com/cometbft/cometbft/proto/tendermint/types"
+	cmttypes "github.com/cometbft/cometbft/types"
+	"github.com/oasisprotocol/oasis-core/go/common/cbor"
+)
+
+// BlockMeta is the CometBFT-specific per-block metadata.
+type BlockMeta struct {
+	// Header is the CometBFT block header.
+	Header *cmttypes.Header `json:"header"`
+	// LastCommit is the CometBFT last commit info.
+	LastCommit *cmttypes.Commit `json:"last_commit"`
+}
+
+// TryUnmarshal attempts to unmarshal the given data into a BlockMeta.
+//
+// It first tries to unmarshal into V1, and if that fails, it tries to
+// unmarshal into V2.
+//
+// We only try to unmarshal into the version of the metadata structure
+// starting at Oasis-Core Eden (V1), and into V2 (starting at #6235).
+// This may fail on blocks from an incompatible earlier version.
+func (b *BlockMeta) TryUnmarshal(data []byte) error {
+	// Try to unmarshal into V1 first.
+	if err := cbor.Unmarshal(data, &b); err == nil {
```

Comment:
I would use `select` here as reader might not see `==`.

Reply:
Updated